lib/bookmark_manager.rb

```diff
@@ -95,8 +89,27 @@ def update(bookmark_id:, name:, reminder_type:, reminder_at:, options: {})
     success
   end
 
+  def toggle_pin(bookmark_id:)
+    bookmark = find_bookmark_and_check_access(bookmark_id)
+    bookmark.pinned = !bookmark.pinned
+    success = bookmark.save
+
+    if bookmark.errors.any?
+      return add_errors_from(bookmark)
+    end
+
+    success
+  end
+
   private
 
+  def find_bookmark_and_check_access(bookmark_id)
+    bookmark = Bookmark.find_by(id: bookmark_id)
+    raise Discourse::NotFound if bookmark.blank?
```

Comment:
why do we need blank? here ?

I think find_by will return nil if not found, so just a !bookmark should be enough no ?

Reply:
Just a habit I guess, I find `bookmark.blank?` more explicit, but I will just change to `!bookmark`